internal/controller/volsync/vshandler.go

```diff
@@ -1112,6 +1118,30 @@ func (v *VSHandler) DeleteRD(pvcName string, pvcNamespace string) error {
 	return nil
 }
 
+func (v *VSHandler) pruneOldSnapshots(pvcNamespace string) error {
```

Comment:
(nit) Maybe a comment is needed for this function. Something like:
```
// pruneOldSnapshots removes older VolumeSnapshots in the given PVC namespace, 
// ensuring that only the most recent snapshot is retained. 
```

Reply:
@BenamarMk done

Reply:
I'll take a look at the unit test failure before merging...